### PR TITLE
Implement tap on iOS using private XCTest API

### PR DESCRIPTION
**Pull request by berikv**
## Proposed Changes

Use `RunnerDaemonProxy().synthesize(eventRecord: tapEvent)` instead of 
```
 xcuiApplication
                .coordinate(withNormalizedOffset: CGVector(dx: 0, dy: 0))
                .withOffset(CGVector(
                    dx: Double(requestBody.x),
                    dy: Double(requestBody.y)
                ))
                .tap()
```

By using the private API it is not needed to look up the XCUIApplication.

### CODE REVIEW

**Review by dmitry-zaitsev — APPROVED** [review]
Awesome!!

File: maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/EventRecord.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import UIKit
+
+class EventRecord {
+    let eventRecord: NSObject
+    var offset = TimeInterval(0)
+    
+    init(orientation: UIInterfaceOrientation) {
+        let name = "Single-Finger Touch Action"
+        // let name = "Multi-Finger Touch Action"
+        eventRecord = objc_lookUpClass("XCSynthesizedEventRecord")?
```
Comment (artem888) [inline maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/EventRecord.swift:11]: ditto

**Review by artem888 — APPROVED** [review]

File: maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/TouchRouteHandler.swift [hunk 2]
```diff
@@ -7,35 +7,28 @@ class TouchRouteHandler : RouteHandler {
...
-                .withOffset(CGVector(
-                    dx: Double(requestBody.x),
-                    dy: Double(requestBody.y)
-                ))
-                .tap()
-            
+        logger.info("Tapping \(requestBody.x), \(requestBody.y)")
+
+        let eventRecord = EventRecord(orientation: .portrait)
+        eventRecord.addPointerTouchEvent(at: CGPoint(x: CGFloat(requestBody.x), y: CGFloat(requestBody.y)))
```
[the remark this answers is not in the document]
Comment (berikv) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/Handlers/TouchRouteHandler.swift:20]: Yes, the api uses positions given in [_points_](https://stackoverflow.com/a/12019236/439096).

File: maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        let clazz: AnyClass = NSClassFromString("XCTRunnerDaemonSession")!
+        let selector = NSSelectorFromString("sharedSession")
+        let imp = clazz.method(for: selector)
+        typealias Method = @convention(c) (AnyClass, Selector) -> NSObject
+        let method = unsafeBitCast(imp, to: Method.self)
+        let session = method(clazz, selector)
+
+        proxy = session
+            .perform(NSSelectorFromString("daemonProxy"))
+            .takeUnretainedValue() as! NSObject
```
[the remark this answers is not in the document]
Comment (berikv) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift:16]: Swift...?

[the remark this answers is not in the document]
Comment (berikv) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift:16]: If proxy.release() is called, XCTest crashes somewhere internally due to over releasing...

[the remark this answers is not in the document]
Comment (berikv) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift:16]: It's not possible to profile the test runner. Having 10-100 proxy objects also wouldn't really be an issue.

[the remark this answers is not in the document]
Comment (berikv) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift:16]: And using `takeRetainedValue()` crashes upon usage

[the remark this answers is not in the document]
Comment (artem888) [reply maestro-driver-ios/maestro-driver-iosUITests/Routes/XCTest/RunnerDaemonProxy.swift:16]: alright, if it is only during the runtime of the runner app, no problem

**Comment by amanjeetsingh150** [comment]
Can you check if these taps are successful for the screens where it was failing?

**Comment by amanjeetsingh150** [comment]
Also just to confirm you have built the runner app with 15.* simulator right?

**Comment by berikv** [comment]
> Also just to confirm you have built the runner app with 15.* simulator right?

I'm running on iOS 15.5. Why did we build using iOS 15.4 actually? Maestro cloud runs on 15.5 currently..

**Comment by berikv** [comment]
> > Also just to confirm you have built the runner app with 15.* simulator right?
> 
> I'm running on iOS 15.5. Why did we build using iOS 15.4 actually? Maestro cloud runs on 15.5 currently..

Rebuild using iOS 15.4 iPhone 13

**Comment by berikv** [comment]
> Can you check if these taps are successful for the screens where it was failing?

I had one screen to test on, that one worked using private API. Do you have another for me to test?

**Comment by artem888** [comment]
just to remind - have you rebased master to your branch before rebuilding? asking, because I forgot to do it once myself and previous changes were ignored :)

**Comment by berikv** [comment]
> just to remind - have you rebased master to your branch before rebuilding? asking, because I forgot to do it once myself and previous changes were ignored :)

Yes, I rebuilt after rebasing

**Comment by amanjeetsingh150** [comment]
> I'm running on iOS 15.5. Why did we build using iOS 15.4 actually? Maestro cloud runs on 15.5 currently..

I agree @artem888 is going to work on this I would lets sync with him and propose him this

**Review by amanjeetsingh150 — APPROVED** [review]

**Pull request merged**